### ISSUE TITLE
Combobox popover unexpectedly closes when clicking Listbox.Option checkbox

### ISSUE THREAD

**Issue by LA1CH3** [Shopify/polaris-react-archive#4844]
## Issue summary

When using the `Combobox` and `Listbox` components, the `Popover` will unexpectedly close when clicking on a `Listbox.Option` checkbox. If you go to the [Combobox docs](https://polaris.shopify.com/components/forms/combobox) with the "Multiple tags autocomplete" example, you will see the expected behavior- clicking anywhere on the `Listbox.Option` (including the checkbox) will select/deselect the option but the Popover remains open. However, if you go to the [Codesandbox](https://codesandbox.io/s/9wkcl?module=App.js) for the same example, you will see the issue- if you open the popover but then click on the Checkbox of one of the `Listbox.Option`, the `Popover` will close. 

In a Slack thread it was mentioned that [PopoverOverlay](https://github.com/Shopify/polaris-react/blob/main/src/components/Popover/components/PopoverOverlay/PopoverOverlay.tsx) might be the culprit:

> TL;DR is that if a click event is determined to not be inside the parent it forces the popover to close. One way this has manifested is if the component unmounts after click the click nodeContainsDescendant check fails (because it's not mounted aka not a child of anything) so I wonder if we're seeing something in the same vain here?

## Expected behavior

Clicking anywhere on a `Listbox.Option` within a `Popover`, the `Popover` should remain open. 

## Actual behavior

If you click on the checkbox within a `Listbox.Option` within a `Popover`, the `Popover` will close. 

<details>
<summary>Popover closing unexpectedly</summary>
<img src="https://user-images.githubusercontent.com/11605788/146387740-3d5b78c2-4748-49ab-b841-2aaa5d6dcdce.gif" alt="Combobox sandbox popover closing when clicking on checkbox" />
</details>

## Steps to reproduce the problem

1. Click/focus on text input
2. Click on a checkbox on one of the popover list items
3. Observe popover closes

## Reduced test case

[Codesandbox](https://codesandbox.io/s/9wkcl?module=App.js)

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 7.5.0
- Browser: Chrome
- Device: Mac M1
- Operating System: MacOS Monterey 12.1

**Comment by ghost** [comment]
👋 Thanks for opening your first issue. A contributor should give feedback soon. If you haven’t already, please check out the [contributing guidelines](https://github.com/Shopify/polaris-react/blob/master/.github/CONTRIBUTING.md).

**Comment by chloerice** [comment]
Fixed in https://github.com/Shopify/polaris/pull/6954

**Comment by V-SpringT** [comment]
I still have an issue on the mobile browser; when the ListBox is inside a Popover, selecting an Option will close the Popover.
